src/EcomDestinationWriter.cs

```diff
@@ -1691,9 +1714,77 @@ private void WritePrices(Dictionary<string, object> row, Dictionary<string, Colu
             LastPriceId = LastPriceId + 1;
             dataRow["PriceID"] = "ImportedPRICE" + LastPriceId;
         }
-        if (!columnMappings.ContainsKey("PriceCurrency"))
+        if (!columnMappings.TryGetValue("PriceCurrency", out var priceCurrencyColumn))
+        {
+            dataRow["PriceCurrency"] = Ecommerce.Services.Currencies.GetDefaultCurrency().Code;
+        }
+        else
         {
-            dataRow["PriceCurrency"] = String.Empty;
+            var priceCurrencyValue = (string)row[priceCurrencyColumn.SourceColumn.Name];
```

Comment:
maybe add a null check on .?SourceColumn?.Name, once the mapping is a Const mapping then a source column was set to None in DW9, so maybe it is the same in DW10 then we need to consider that the source column can be null